scripts/nbconverted/main_Pa_sim_enhance_AtoB.py

```diff
@@ -0,0 +1,225 @@
+    new_A_exp = random.uniform(0.4, 1.0)
+    
+    # Set gene expression value for genes in set A to be the same random value selected
+    data.loc[sample_id][geneSetA] = new_A_exp
+    
+    # Select sample gene from set A to be representative since the expression is the same
+    # for all genes in the set
+    geneSetA_pick = geneSetA[0]
+    
+    # Check if expression of genes in set A exceed the threshold
+    if data.loc[sample_id,geneSetA_pick] > thresholdA:
+        
+        # Scale genes by some fixed percentage
+        for gene in geneSetB:
+            data.loc[sample_id,gene] = (1+effect_size)*data.loc[sample_id,gene]            
+            
+# if any exceed 1 then set to 1 since gene expression is normalized
+data[data>=1.0] = 1.0
+
+
+# In[11]:
+
+
+# Dataframe with only genes in set A
+geneA_only = pd.DataFrame(data[geneSetA_pick], index=data.index, columns=[geneSetA_pick])
+
+# Drop genes in set A
+data_holdout = data.drop(geneSetA, axis=1)
+
+geneA_only
+
+
+# In[12]:
+
+
+# checkpoint:  Check that the holdout set does not include genes in set A
+assert(data_holdout.shape[1] == (data.shape[1]-len(geneSetA)))
+print("Confirmed that holdout set is the correct size")
+
+
+# In[13]:
+
+
+# Create list of base directories
+base_dirs = [os.path.join(os.path.dirname(os.getcwd()), 'data'),
+             os.path.join(os.path.dirname(os.getcwd()), 'encoded'),
+             os.path.join(os.path.dirname(os.getcwd()), 'models'),
+             os.path.join(os.path.dirname(os.getcwd()), 'output'),
+             os.path.join(os.path.dirname(os.getcwd()), 'stats'),
+             os.path.join(os.path.dirname(os.getcwd()), 'viz')    
+]
+
+# Check if directory exist otherwise create
+for each_dir in base_dirs:
+    analysis_dir = os.path.join(each_dir, analysis_name)
+    if os.path.exists(analysis_dir):
+        print('directory already exists: {}'.format(analysis_dir))
+    else:
+        os.mkdir(analysis_dir)
+        print('creating new directory: {}'.format(analysis_dir))
+
+
+# In[14]:
+
+
+# Output the new gene expression values for each sample
+train_input_file = os.path.join(base_dirs[0], analysis_name, "train_model_input.txt.xz")
+data_holdout.to_csv(train_input_file, sep='\t', compression='xz', float_format="%.5g")
+
+# Output log file with params
+log_file = os.path.join(os.path.dirname(os.getcwd()), 'metadata', analysis_name + '.txt')
+
+args_dict = {
+    "size of gene set A": geneSetA_size,
+    "size of gene set B": geneSetB_size,
+    "genes in set B": geneSetB,
+    "effect size of genes in set B": effect_size,
+    "threshold of A activation": thresholdA,
+}
+
+with open(log_file, 'w') as f:
+    for key, value in args_dict.items():
+        f.write('%s: %s\n' % (key, value))
+        
+# Output geneA only file using sample gene A selected
+geneA_file = os.path.join(base_dirs[0], analysis_name, geneSetA_pick + ".txt")
+geneA_only.to_csv(geneA_file, sep='\t', float_format="%.5g")
+
+
+# In[15]:
```

Comment:
I am having a hard time understanding what is happening in this cell. Can it be split into multiple cells?

The plots are representing correlation in the latent space between input and output? For PCA, VAE and raw expression? What is being interpolated?

I also don't remember many of the functions added here. Maybe would be good to remind someone what the goal of simuation study is? What is the hypothesis?

Reply:
Very good advice!  I have added additional comments to describe the experiment.  Hopefully this makes more sense.  It also brought to light the fact that I was performing the simulation experiment incorrectly!  I am re-running the corrected scripts now.  

I have also removed the "redo" extension on some of the files.  This was also accidentally left in during testing.